### ISSUE TITLE
Prevent deployment when the target branch is not the default branch

### ISSUE THREAD

**Issue by GrantBirki** [github/branch-deploy#340]
Currently, you are able to open a pull request that is targeting a branch other than the default (i.e. `main`) and you can deploy that branch while bypassing branch protection rules.

For example, you might have a branch protection rule on `main` that says a PR must be approved before it can be deployed. If you simply open a PR that targets a branch other than main, you would be able to deploy those changes without an approval.

Example:

```mermaid
gitGraph
    commit
    commit
    branch develop
    checkout develop
    commit
    commit
    branch bug-fix
    commit id: "fix error" type: HIGHLIGHT
```

In this example, you could not deploy any of the commits from the `develop` branch as they are trying to merge into the `main` branch which has a branch protection rule requiring passing CI, approvals, etc. However, if you open a PR from the `bug-fix` branch, you could `.deploy` the `fix error` commit without issues since it is not targeting a branch with branch protection (or rulesets).

This is a bug and it should be resolved.

**Comment by S3bb1** [comment]
HI @GrantBirki 

We just ran into your "bug" but on our side its intended behavior that we allow deploying from these branches to specific environments. Is there a setting to disable this error?

**Comment by GrantBirki** [comment]
@S3bb1 would you be able to go into more details around how/why you are using this deployment pattern?

Being able to bypass checks, approvals, and branch protection settings is considered very dangerous.

There is currently not a way to disable the safety checks that I added but that could be changed.

**Comment by GrantBirki** [comment]
@S3bb1 Do you have some sort of branch protections or rulesets applied to these "bigger feature" branches? If so, then that would make your workflow a lot more "safe". 

Either way, I will look into making an opt-out option for your use case today as I did not indent on this initial change to be breaking in this way for folks.

**Comment by GrantBirki** [comment]
@S3bb1 I have merged https://github.com/github/branch-deploy/pull/361 which contains changes to disable the logic that is blocking your team.

The latest pre-release is available as well: [`v10.2.0`](https://github.com/github/branch-deploy/releases/tag/v10.2.0).

Please go ahead and test it out and let me know if it works for you. Thanks!
